test/unit/test_report_unit.py

```diff
@@ -0,0 +1,413 @@
+"""Unit tests for the report command."""
+
+from io import StringIO
+from unittest import mock
+
+from util import get_mock_State_and_config
+
+from wily import format_date as fd
+from wily.commands.report import report
+from wily.defaults import DEFAULT_GRID_STYLE
+from wily.helper.custom_enums import ReportFormat
+
+EXPECTED = f"""
+╒════════════╤════════════════╤════════════╤═════════════════╕
+│ Revision   │ Author         │ Date       │ Lines of Code   │
+╞════════════╪════════════════╪════════════╪═════════════════╡
+│ abcdef0    │ Author 0       │ {fd(0)} │ 0 (\u001b[33m-1\u001b[0m)          │
+├────────────┼────────────────┼────────────┼─────────────────┤
+│ abcdef1    │ Author 1       │ {fd(1)} │ 1 (\u001b[33m-1\u001b[0m)          │
+├────────────┼────────────────┼────────────┼─────────────────┤
+│ abcdef2    │ Author 2       │ {fd(2)} │ 2 (\u001b[33m-1\u001b[0m)          │
+├────────────┼────────────────┼────────────┼─────────────────┤
+│ abcdeff    │ Author Someone │ {fd(10)} │ 3 (\u001b[33m-1\u001b[0m)          │
+├────────────┼────────────────┼────────────┼─────────────────┤
+│ abcdeff    │ Author Someone │ {fd(10)} │ 4 (\u001b[33m+1\u001b[0m)          │
+├────────────┼────────────────┼────────────┼─────────────────┤
+│ abcdeff    │ Author Someone │ {fd(10)} │ 3 (0)           │
+├────────────┼────────────────┼────────────┼─────────────────┤
+│ abcdeff    │ Author Someone │ {fd(10)} │ 3 (0)           │
+╘════════════╧════════════════╧════════════╧═════════════════╛
+"""
+EXPECTED = EXPECTED[1:]
+
+
+def test_report_no_message(capsys):
+    path = "test.py"
+    metrics = ("raw.loc",)
+    format = "CONSOLE"
+    mock_State, mock_config = get_mock_State_and_config(3)
+
+    with mock.patch("wily.commands.report.State", mock_State):
+        report(
+            config=mock_config,
+            path=path,
+            metrics=metrics,
+            n=10,
+            output=None,
```

Comment:
output is not optional, so either the types are wrong, or this should fail

Reply:
Done, using `Path()`.